users/luescher/setups/hybrid/gmm_system.py

```diff
@@ -0,0 +1,845 @@
+
+        self.default_align_keep_values = {
+            "default": 5,
+            "selected": gs.JOB_DEFAULT_KEEP_VALUE,
+        }
+
+    # -------------------- Setup --------------------
+    def init_system(
+        self,
+        gmm_args: GmmPipelineArgs,
+        train_data: Dict[str, GmmDataInput],
+        dev_data: Dict[str, GmmDataInput],
+        test_data: Dict[str, GmmDataInput],
+    ):
+        """
+        :param gmm_args: parameters for the different Gmm-HMM steps
+        :param train_data: dict(str: GmmInput)
+        :param dev_data: dict(str: GmmInput)
+        :param test_data: dict(str: GmmInput)
+        :return:
+        """
+        self.gmm_args = gmm_args
+        self._init_am(**gmm_args.am_args)
+        for name, v in sorted(train_data.items()):
+            self.corpora[name] = v.corpus_object
+            self.concurrent[name] = v.concurrent
+            self._init_corpora(name)
+            self._init_lexica(name, **v.lexicon)
+            self.train_corpora.append(name)
+
+        for name, v in sorted(dev_data.items()):
+            self.corpora[name] = v.corpus_object
+            self.concurrent[name] = v.concurrent
+            self._init_corpora(name)
+            self._init_lexica(name, **v.lexicon)
+            self._init_lm(name, **v.lm)
+            self.dev_corpora.append(name)
+
+        for name, v in sorted(test_data.items()):
+            self.corpora[name] = v.corpus_object
+            self.concurrent[name] = v.concurrent
+            self._init_corpora(name)
+            self._init_lexica(name, **v.lexicon)
+            self._init_lm(name, **v.lm)
+            self.test_corpora.append(name)
+        self.cart_questions = gmm_args.cart_questions
+
+    @tk.block()
+    def _init_am(self, **kwargs):
+        self.crp["base"].acoustic_model_config = am.acoustic_model_config(**kwargs)
+
+    @tk.block()
+    def _init_corpora(self, name):
+        segm_corpus_job = corpus_recipes.SegmentCorpusJob(
+            self.corpora[name].corpus_file, self.concurrent[name]
+        )
+        self.set_corpus(
+            name=name,
+            corpus=self.corpora[name],
+            concurrent=self.concurrent[name],
+            segment_path=segm_corpus_job.segment_path,
+        )
+        self.jobs[name]["segment_corpus"] = segm_corpus_job
+
+    @tk.block()
+    def _init_lm(self, name, file, type, scale, **kwargs):
+        self.crp[name].language_model_config = rasr.RasrConfig()
+        self.crp[name].language_model_config.type = type
+        self.crp[name].language_model_config.file = file
+        self.crp[name].language_model_config.scale = scale
+
+    @tk.block()
+    def _init_lexica(self, name, file, normalize_pronunciation, **kwargs):
+        self.crp[name].lexicon_config = rasr.RasrConfig()
+        self.crp[name].lexicon_config.file = file
+        self.crp[name].lexicon_config.normalize_pronunciation = normalize_pronunciation
+
+    def allow_zero_weights(self, name: str = "base"):
```

Comment:
This can also be done with the training args for the individual training steps.. `{mono,tri,...}_args`
`init_am` should also be possible I think.. not 100% sure.. maybe @curufinwe knows?
so remove completely?

Reply:
merged into `_init_am` function. moved from `acoustic_model_post_config` to `acoustic_model_config`, so has to be set correctly from the start, otherwise set `allow_zero_weights` via `{mono,tri,...}_args`